frappe/email/doctype/notification/notification.py

```diff
@@ -94,8 +94,18 @@ def on_update(self):
 		frappe.cache.hdel("notifications", self.document_type)
 		path = export_module_json(self, self.is_standard, self.module)
 		if path:
-			# js
-			if not os.path.exists(path + ".md") and not os.path.exists(path + ".html"):
+			# template
+			if os.path.exists(path + ".md"):
+				with open(path + ".md", "w") as f:
+					f.write(self.message)
+			elif os.path.exists(path + ".html"):
+				with open(path + ".html", "w") as f:
+					f.write(self.message)
+			elif os.path.exists(path + ".txt"):
+				with open(path + ".txt", "w") as f:
+					f.write(self.message)
+			# first write as markdown (can contain html)
+			else:
 				with open(path + ".md", "w") as f:
 					f.write(self.message)
```

Comment:
```suggestion
			formats = [".md", ".html", ".txt"]
			use_default = True

			for ext in formats:
				file_path = path + ext
				if os.path.exists(file_path):
					with open(file_path, "w") as f:
						f.write(self.message)
					use_default = False
					break

			if use_default:
				with open(path + ".md", "w") as f:
					f.write(self.message)
```

Reply:
Nice, thanks for the prompt. That could even be made:

```python
			formats = [".md", ".html", ".txt"]

			for ext in formats:
				file_path = path + ext
				if os.path.exists(file_path):
					with open(file_path, "w") as f:
						f.write(self.message)
					break

			else:
				with open(path + ".md", "w") as f:
					f.write(self.message)
```

Reply:
Oooh…
`for… break… else`
Nice! 😄